src/solver/simulation/hydro-final-reservoir-level-functions.cpp

```diff
@@ -31,34 +31,38 @@
 namespace Antares::Solver
 {
 
-void prepareFinalReservoirLevelDataPerMcY(Data::Study& study, uint year)
+void initializeFinalLevelData(Data::Study& study)
 {
-    study.areas.each(
-      [&study, &year](Data::Area& area)
-      {
-          auto& finalInflows = area.hydro.finalLevelInflowsModifier;
-          auto& scenarioInitialHydroLevels = study.scenarioInitialHydroLevels;
-          auto& scenarioFinalHydroLevels = study.scenarioFinalHydroLevels;
-          auto& parameters = study.parameters;
+    study.areas.each([&study](Data::Area& area)
+    {
+        area.hydro.finalLevelInflowsModifier.initialize(study.scenarioInitialHydroLevels,
+                                                        study.scenarioFinalHydroLevels,
+                                                        study.parameters.simulationDays.end,
+                                                        study.parameters.nbYears);
+    });
+}
 
-          finalInflows.initializeData(
-            scenarioInitialHydroLevels, scenarioFinalHydroLevels, parameters, year);
+void CheckFinalReservoirLevelsForYear(Data::Study& study, uint year)
+{
+    study.areas.each([&study, &year](Data::Area& area)
+    {
+        if (!area.hydro.finalLevelInflowsModifier.CheckInfeasibility(year))
+        {
+            AntaresSolverEmergencyShutdown();
```

Comment:
This function was removed by @sylvlecl, please `throw` instead.

Reply:
Done